### PR TITLE
Re-partition MSBuild test groups

### PR DESCRIPTION
**Pull request by echesakovMSFT**
This changes a tests paritioning scheme of https://github.com/dotnet/coreclr/pull/17161 in several different ways:
1. **Make the test groups are of equal (to some extent) sizes** keeping the number of tests in one group less than 1000;
2. As a resulf of 1) this increases a number of groups needed for Pri0 tests to 3 and descreases a number of groups for Pri1 tests to 10;
3. This also **changes a way of tests partitioning scheme is defined** - instead of explicitly specifying MSBuild Include-patterns this defines a boundaries between test groups in a form of _StartsWith ItemGroup. Then I use StringComparer.Ordinal to pick tests that belong to a particular group.

All of this was done to mitigate https://github.com/dotnet/coreclr/issues/20236 which I believe caused by OOM due to enormous number of projects loaded during build-tests (same reasoning as in https://github.com/dotnet/coreclr/pull/17161).

Also fixes https://github.com/dotnet/coreclr/issues/22627 and should prevent from such issues in the future.

I want to mention that I was considering using automatic scheme before converging back to the manual one. 
There are several problem with automatic partitioning that in theory can be solved but would take too much effort to do this and probably can be postponed:
1. The priority of a test (and whether the test is going to be built) is not known during dirs.proj time - we need to import a test first to figure this out;
2. If we ignore 1) and make an assumption that pri0 and pri1 tests are uniformly distributed than we can do different tricks with partitioning scheme. For example, *Group n* can composed by taking *every n-th test*. However, I don't know about a way to do this in MSBuild other than implementing logic in C# MSBuild task. It could be implemented using regexps (and I have a prototype doing this) but the code would look so cryptic that I decided not do this;
3. The proper way of doing automatic partitioning should be based on some hash function (which is naturally uniformly distributed). As an example, the hash can be computed based on a test file name as input. However, the hash function directly available in MSBuild is String.GetHashCode is not stable - meaning the distribution would change from run to run adding non-determinism into build system.

@dotnet/jit-contrib

### CODE REVIEW

**Comment by echesakovMSFT** [comment]
@AaronRobinsonMSFT  Can you please take a look at the MSBuild changes ?

File: tests/src/dirs.proj [hunk 1]
```diff
@@ -33,218 +33,86 @@
...
-      <Project Include="*\**\*.ilproj" Exclude="@(DisabledProjects)">
-        <AdditionalProperties>OSGroup=$(OSGroup)</AdditionalProperties>
-      </Project>
+      <_GroupStartsWith Include="JIT\Regression\CLR-x86-JIT\V1-M12-Beta2\b35351\b35351.ilproj">
+        <GroupNumber>3</GroupNumber>
+      </_GroupStartsWith>
     </ItemGroup>
 
     <!-- Test build is divided in slices which can be created within Test Group
          Priority 1 or higher tests are build using Test Group 2 with 16 subgroups or slices -->
```
Comment (AaronRobinsonMSFT) [inline tests/src/dirs.proj]: Update 16 to 10.

File: tests/src/dirs.proj [hunk 2]
```diff
@@ -33,218 +33,86 @@
...
+
     <!-- Test build is divided in slices which can be created within Test Group
          Priority 0 tests are build using Test Group 1 with 2 subgroups or slices -->
 
-    <ItemGroup Condition="$(__TestGroupToBuild) == '1' And $(TestBuildSlice) == '1'">
-      <Project Include="*\**\*.csproj" Exclude="@(DisabledProjects)">
-        <AdditionalProperties>OSGroup=$(OSGroup)</AdditionalProperties>
-      </Project>
-    </ItemGroup>
+    <ItemGroup Condition=" '$(CLRTestPriorityToBuild)' == '0' ">
```
Comment (AaronRobinsonMSFT) [inline tests/src/dirs.proj]: Comment what group one is, no need to define since it would be very hard to make that work.

**Comment by echesakovMSFT** [comment]
Both **coreclr-ci** and **coreclr-outerloop** have succeeded. 
Ignoring Jenkins infra failures.

**Pull request merged**